db_assessment/import_db_assessment.py

```diff
@@ -662,4 +690,103 @@ def insertErrors(invalidfiles,op_df,gcpProjectName,bq_dataset):
+    tmpdataFrame=pd.DataFrame()
+
+    if btsource=='fromimportDataframeToBQ':
+
+        if dataframeornot is not None:
+            if 'PKEY' in df.columns.to_list():
+                # df2.reset_index(inplace=True)
+                if 'dbsizing_summary' not in tableName: # Need to add code to support dbsizing_summary or fix the issue with it 
+                    pkeygroupby=df.groupby(['PKEY']).size()
+                    # pkeygroupby=df2.groupby(level=0).size()
+                    pkeycount=pkeygroupby.to_dict()
+                    for pkeyname,pkeyname_rowcount in pkeycount.items():
+                        if 'Elapsed' not in pkeyname:
+                            # twoDimensionalList.append([tableName,pkeyname,'SUCCESS',pkeyname_rowcount])    
+                            tmpdataFramedict = {"Target Table":tableName,"Distinct Pkey":pkeyname,"Import Status":"SUCCESS","Loaded rows":pkeyname_rowcount}
+                            tmpdataFrame = tmpdataFrame.append(tmpdataFramedict, ignore_index = True)  
+                            if len(tmpdataFrame) >0:
+                                importresults = pd.concat([importresults, tmpdataFrame], ignore_index = True, axis = 0)
+
+    elif btsource=='fromimportCSVToBQ':
+        fileName=invalidfiles
+        if 'opdbt' not in fileName: # to go past the error seen during Demo to Eri on Apr 26, 2022
+            if rowsimported >=0:
+                tmpdataFramedict = {"Target Table":tableName,"Distinct Pkey":getPkeyFromFile(fileName),"Import Status":"SUCCESS","Loaded rows":rowsimported}
+                tmpdataFrame = tmpdataFrame.append(tmpdataFramedict, ignore_index = True)  
+
+            else:
+                tmpdataFramedict = {"Target Table":tableName,"Distinct Pkey":getPkeyFromFile(fileName),"Import Status":"FAILED","Loaded rows":0}
+                tmpdataFrame = tmpdataFrame.append(tmpdataFramedict, ignore_index = True)
+            if len(tmpdataFrame) >0:
+                importresults = pd.concat([importresults, tmpdataFrame], ignore_index = True, axis = 0)
+  
+    elif btsource=='invalidfiles':
+        for fileName, error in invalidfiles.items():
+            tmpdataFramedict = {"Target Table":getObjNameFromFiles(fileName,'__',1),"Distinct Pkey":getPkeyFromFile(fileName),"Import Status":"FAILED","Loaded rows":0}
+            tmpdataFrame = tmpdataFrame.append(tmpdataFramedict, ignore_index = True)
+            if len(tmpdataFrame) >0:
+                importresults = pd.concat([importresults, tmpdataFrame], ignore_index = True, axis = 0)
+
+    
+    # if 'opdbt'  in fileName:
+    #     importresults=importresults
+    # else:
+    #     if len(tmpdataFrame) >0:
+    #         importresults = pd.concat([importresults, tmpdataFrame], ignore_index = True, axis = 0)
+
+    return importresults
+
+def processBTResults(importresults):
+
+    #Create a in-memory database and connection
+    #Database is used for 
+    conn=sqlite3.connect(':memory:')
+    c=conn.cursor()
```

Comment:
Let's change it to pandas later

Reply:
changed to pandas in latest commit for bug132 branch